resource/config.go

```diff
@@ -175,24 +176,30 @@ func (assoc AssociatedResourceConfig) MarshalJSON() ([]byte, error) {
 // Equals checks if the two configs are deeply equal to each other. Validation
 // related fields and implicit dependencies will be ignored.
 func (conf Config) Equals(other Config) bool {
+	for association, assocCfg := range conf.AssociatedAttributes {
+		otherAssocCfg, ok := other.AssociatedAttributes[association]
+		if !ok || !assocCfg.Equals(otherAssocCfg) {
+			return false
+		}
+	}
+
 	// These `Config` objects are copies. Changing the members here for equality checking does not
 	// impact the original versions.
 	conf.alreadyValidated = false
 	conf.ImplicitDependsOn = nil
 	conf.cachedImplicitDeps = nil
 	conf.cachedErr = nil
+	conf.ConvertedAttributes = nil
+	conf.AssociatedResourceConfigs = nil
+	conf.AssociatedAttributes = nil
+
 	other.alreadyValidated = false
 	other.ImplicitDependsOn = nil
 	other.cachedImplicitDeps = nil
 	other.cachedErr = nil
-
-	// TODO(RSDK-5523): Once builtin datamanagers' AssociatedConfigLinkers no
-	// longer rely on appensions to the service's ConvertedAttributes field,
-	// continue ignoring ConvertedAttributes for equality for all resources.
-	if conf.API != APINamespaceRDK.WithServiceType("data_manager") {
-		conf.ConvertedAttributes = nil
-		other.ConvertedAttributes = nil
-	}
+	other.ConvertedAttributes = nil
+	other.AssociatedResourceConfigs = nil
+	other.AssociatedAttributes = nil
```

Comment:
Its kinda confusing to nil out all these things.  I think we could alternatively just be manually checking equality for all the fields we actually care about

Reply:
it's 50/50 at this point, but manually checking is harder to maintain

Reply:
Nilling out seems fine to me given the precedence of that pattern all across our config `Equals`s methods.